### PR TITLE
Makefile: Cleanup, alpine and amd64 only UDP

### PR DESCRIPTION
**Pull request by tomdee**
Reorganize the Makefile to have different sections for build, test,
release and development.

Move to using Alpine linux as the base image. This means we no longer
need to build iptables and it makes it easier to install other packages
in future (e.g. strongswan).

To allow alpine to be used, static binaries for all platforms are
needed. It's not possible to create these for the UDP backend for
non-amd64 platforms. This is because it relies on CGO. So this PR makes
the UDP backend work on amd64 only.

### CODE REVIEW

File: Dockerfile.amd64 [hunk 1]
```diff
@@ -1,12 +1,11 @@
-FROM frolvlad/alpine-glibc
+FROM alpine
 
 MAINTAINER Tom Denham <tom@tigera.io>
```
Comment (gunjan5) [inline Dockerfile.amd64]: nit: `MAINTAINER` has been deprecated, we should probably switch to `LABEL`

File: Makefile [hunk 2]
```diff
@@ -6,40 +6,61 @@ REGISTRY?=quay.io/coreos/flannel
...
 ARCH?=amd64
+# Only enable CGO (and build the UDP backend) on AMD64
+ifeq ($(ARCH),amd64)
+	CGO_ENABLED=1
+else
+	CGO_ENABLED=0
+endif
+
+# Go version to use for builds
+GO_VERSION=1.8.3
```
Comment (gunjan5) [inline Makefile]: any reason we're not using Go 1.9.2?

File: backend/udp/udp_amd64.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
+// Copyright 2015 flannel authors
```
Comment (gunjan5) [inline backend/udp/udp_amd64.go]: 2017

File: backend/udp/udp_amd64.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package udp
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"golang.org/x/net/context"
```
Comment (gunjan5) [inline backend/udp/udp_amd64.go]: context has been move to `"context"` as part of the std lib since Go 1.7

Comment (tomdee) [reply backend/udp/udp_amd64.go]: Sure, but it hasn't been updated in flannel yet and that change shouldn't be done in this PR

File: Makefile (hunk 2 above)
Comment (tomdee) [reply Makefile]: I don't want to introduce too many changes in this PR. I'll rev the Go version in a different PR

File: Dockerfile.amd64 (hunk 1 above)
Comment (tomdee) [reply Dockerfile.amd64]: Already done in another PR :smile:

**Pull request merged**